File: Csharp.Controller.Api/Routes/StudentRoutes.cs
using Csharp.Controller.Api.Controllers;

namespace Csharp.Controller.Api.Routes;

/// <summary>
/// Defines the routes to the student endpoints.
/// </summary>
internal static class StudentRoutes
{
    /// <summary>
    /// The route to <see cref="StudentsController.Add"/>.
    /// </summary>
    public const string Add = Base;
    
    /// <summary>
    /// The base route to access the endpoints
    /// of the <see cref="StudentsController"/>.
    /// </summary>
    public const string Base = "/api/Students";
    
    /// <summary>
    /// The route to <see cref="StudentsController.Delete"/>. 
    /// </summary>
    public const string Delete = Get;
    
    /// <summary>
    /// The route to <see cref="StudentsController.Get"/>.
    /// </summary>
    public const string Get = $"{Base}/{{id:guid}}";

    /// <summary>
    /// The route to <see cref="StudentsController.GetAll"/>.
    /// </summary>
    public const string GetAll = Base;

    /// <summary>
    /// The route to <see cref="StudentsController.Update"/>.
    /// </summary>
    public const string Update = Get;
}

File: Fsharp.Controller.Api/Routes/StudentRoutes.fs
namespace Fsharp.Controller.Api.Routes

/// <summary>
/// Defines the routes to the student endpoints.
/// </summary>
[<RequireQualifiedAccess>]
module StudentRoutes =

    /// <summary>
    /// The base route to access the endpoints
    /// of the StudentsController.
    /// </summary>
    [<Literal>]
    let Base = "/api/Students"
    
    /// <summary>
    /// The route to StudentsController.Add.
    /// </summary>
    [<Literal>]
    let Add = Base
    
    /// <summary>
    /// The route to StudentsController.Get.
    /// </summary>
    [<Literal>]
    let Get = Base + "/{id:guid}"
    
    /// <summary>
    /// The route to StudentsController.GetAll.
    /// </summary>
    [<Literal>]
    let GetAll = Base
    
    /// <summary>
    /// The route to StudentsController.Delete.
    /// </summary>
    [<Literal>]
    let Delete = Get
    
    /// <summary>
    /// The route to StudentsController.Update.
    /// </summary>
    [<Literal>]
    let Update = Get